### PR TITLE
Add missing artifacts for reuse compliance

### PR DESCRIPTION
**Pull request by a-thaler**
**Description**

Changes proposed in this pull request:

- Add license and reuse status badge
- ...
- ...

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/18565
https://github.com/kyma-project/kyma/issues/18593
https://github.com/kyma-project/kyma/issues/18594

### CODE REVIEW

File: LICENSES/Apache-2.0.txt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+END OF TERMS AND CONDITIONS
+
+APPENDIX: How to apply the Apache License to your work.
+
+To apply the Apache License to your work, attach the following boilerplate notice, with the fields enclosed by brackets "[]" replaced with your own identifying information. (Don't include the brackets!)  The text should be enclosed in the appropriate comment syntax for the file format. We also recommend that a file or class name and description of purpose be included on the same "printed page" as the copyright notice for easier identification within third-party archives.
+
+Copyright [yyyy] [name of copyright owner]
```
Comment (mmitoraj) [inline LICENSES/Apache-2.0.txt:61]: Please update

File: .reuse/dep5 [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ project’s code for that SAP External Product, subject to the terms of such
+ license agreement. If you do not have a valid license agreement for the use of
+ a particular SAP External Product, then you may only make use of any API Calls
+ in this project for that SAP External Product for your internal, non-productive
+ and non-commercial test and evaluation of such API Calls. Nothing herein grants
+ you any rights to use or access any SAP External Product, or provide any third
+ parties the right to use of access any SAP External Product, through API Calls.
+
+Files: *
+Copyright: 2023 SAP SE or an SAP affiliate company and Kyma contributors
```
Comment (mmitoraj) [inline .reuse/dep5:28]: Shouldn't it be 2024 already?

Comment (a-thaler) [inline .reuse/dep5:28]:
```suggestion
Copyright: 2024 SAP SE or an SAP affiliate company and Kyma contributors
```

File: LICENSES/Apache-2.0.txt (hunk 1 above)
Comment (a-thaler) [reply LICENSES/Apache-2.0.txt:61]: I see it as a general description on how to apply

Comment (a-thaler) [inline LICENSES/Apache-2.0.txt:61]:
```suggestion
Copyright: 2024 SAP SE or an SAP affiliate company and Kyma contributors
```

**Review by mmitoraj — APPROVED** [review]

**Review by pbochynski — APPROVED** [review]

**Pull request merged**